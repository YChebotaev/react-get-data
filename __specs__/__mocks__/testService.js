export const makeTestService = ({ testResult, testError }) => ({
  testMethod: jest.fn(() => ({ testResult })),
  asyncMethod: jest.fn(() => Promise.resolve({ testResult })),
  asyncError: jest.fn(() => Promise.reject(testError))
})
