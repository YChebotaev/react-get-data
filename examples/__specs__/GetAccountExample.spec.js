import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import { account, brokenAccount } from './__mocks__/accountService'
import { GetAccountExample } from '../GetAccountExample'

describe('<GetAccountExample>', () => {
  it('should render spinner', () => {
    const getAccountExample = ReactTestRenderer.create(
      <GetAccountExample account={account} />
    )
    expect(getAccountExample.toJSON()).toMatchSnapshot()
  })

  it('should display account', done => {
    const getAccountExample = ReactTestRenderer.create(
      <GetAccountExample account={account} />
    )
    setTimeout(() => {
      expect(getAccountExample.toJSON()).toMatchSnapshot()
      done()
    }, 600)
  })

  it('should display error', done => {
    const getAccountExample = ReactTestRenderer.create(
      <GetAccountExample account={brokenAccount} />
    )
    setTimeout(() => {
      expect(getAccountExample.toJSON()).toMatchSnapshot()
      done()
    }, 600)
  })
})
