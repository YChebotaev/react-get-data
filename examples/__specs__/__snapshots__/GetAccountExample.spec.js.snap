// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetAccountExample> should display account 1`] = `
<div
  className="account"
/>
`;

exports[`<GetAccountExample> should display error 1`] = `
<div
  className="error"
>
  Broken account
</div>
`;

exports[`<GetAccountExample> should render spinner 1`] = `
<div
  className="spinner"
/>
`;
